static/app/views/alerts/rules/issue/ticketRuleModal.tsx

```diff
@@ -169,7 +169,16 @@ class TicketRuleModal extends AbstractExternalIssueForm<Props, State> {
         .map(field => {
           // Overwrite defaults from cache.
           if (instance.hasOwnProperty(field.name)) {
-            field.default = instance[field.name] || field.default;
+            // It's possible that the choices field was not restored to its
+            // prior state if the field has ignorePriorChoices set to true, so
+            // we need to check that the prior value exists in the choices.
+            if (
+              instance[field.name] &&
+              field.choices &&
+              field.choices.some(item => item[0] === instance[field.name])
```

Comment:
`field.choices` takes the form `[("1", "Proj1"), ("2", "Proj2")]` and `instance[field.name]` is always in the form of the first item in the tuple, like `"1"`